scheduler/src/cook/kubernetes/api.clj

```diff
@@ -574,6 +582,7 @@
         {:keys [custom-shell default-checkpoint-config init-container set-container-cpu-limit? sidecar]}
         (config/kubernetes)
         checkpoint (when checkpoint (merge default-checkpoint-config (util/job-ent->checkpoint job)))
+        checkpoint-memory-overhead (:memory-overhead checkpoint)
```

Comment:
Do we need to account for this when sending synthetic pods too?

https://github.com/twosigma/Cook/blob/master/scheduler/src/cook/kubernetes/compute_cluster.clj#L359-L367

Reply:
No. The synthetic pod is not enabling checkpointing.

Reply:
But doesn't the synthetic pod need to ask for the "extra" memory? Just like it asks for the extra resources for the sidecar even though it doesn't fire up the sidecar.

Reply:
oops. good point. we shouldn't have added the sidecar-resource-requirements for the synthetic pod. It should have used the same codepath as the scheduler. I'll fix

Reply:
Thanks, sounds good

Reply:
Is your fix going to make #1528 moot?

Reply:
yes. updated